### PR TITLE
Typescript Example App

### PR DESCRIPTION
**Pull request by rgomezp**
This PR includes a brand new example app for React Native built using Typescript. It also deletes the previous example app from the `examples` directory.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,15 +15,15 @@ examples/CocoapodsDemo/ios/CocoapodsDemo.xcodeproj/xcuserdata
...
-examples/RNOneSignal/android/app/build
-examples/RNOneSignal/android/.gradle
-examples/RNOneSignal/android/.idea
-examples/RNOneSignal/android/build
-examples/RNOneSignal/ios/RNOneSignal.xcodeproj/project.xcworkspace/xcshareddata
-examples/RNOneSignal/ios/RNOneSignal.xcodeproj/project.xcworkspace/xcuserdata
-examples/RNOneSignal/ios/RNOneSignal.xcodeproj/xcuserdata
-examples/RNOneSignal/ios/RNOneSignal.xcworkspace/xcshareddata
-examples/RNOneSignal/ios/RNOneSignal.xcworkspace/xcuserdata
+examples/RNOneSignalTS/android/app/build
```
Comment (Jeasmine) [inline .gitignore:18]: do we intend to replace the other example project?

**Review by Jeasmine — COMMENTED** [review]
Even if the app is an example app 

examples/RNOneSignalTS/android/app/debug.keystore shouldn't be pushed

For the Location to work, we need to add the permission to the manifest and the location dependency. Also as an improvement we might want to change the button from SHARE LOCATION to UNSHARE LOCATION since the app starts with share location enable by default.

File: .gitignore (hunk 1 above)
Comment (rgomezp) [reply .gitignore:18]: yes, after discussing with @jkasten2 we decided to remove the other one

File: .gitignore [hunk 2]
```diff
@@ -15,15 +15,16 @@ examples/CocoapodsDemo/ios/CocoapodsDemo.xcodeproj/xcuserdata
...
+examples/RNOneSignalTS/android/app/debug.keystore
```
Comment (rgomezp) [inline .gitignore:19]: added here

**Review by Jeasmine — COMMENTED** [review]
examples/RNOneSignalTS/android/app/debug.keystore is ignored but still added on the commits. Also, the location config shouldn't be added to the example app?

**Comment by rgomezp** [comment]
> examples/RNOneSignalTS/android/app/debug.keystore is ignored but still added on the commits. Also, the location config shouldn't be added to the example app?

Updated

**Pull request merged**